weld/codegen/llvm2/numeric.rs

```diff
@@ -262,13 +264,41 @@ impl NumericExpressionGen for LlvmGenerator {
                             } else {
                                 LLVMBuildNot(ctx.builder, equal, c_str!(""))
                             }
-                        }
+                        },
+                        BinOpKind::LessThan | BinOpKind::GreaterThanOrEqual => {
+                            use super::cmp::GenCmp;
+                            let func = self.gen_cmp_fn(ty)?;
+                            let mut args = [ctx.get_value(left)?, ctx.get_value(right)?];
+                            let cmp = LLVMBuildCall(ctx.builder, func, args.as_mut_ptr(), args.len() as u32, c_str!(""));
+                            let lt = LLVMBuildICmp(ctx.builder, LLVMIntSLT, cmp,
+                                                   self.i32(0), c_str!(""));
+                            
+                            if op == BinOpKind::LessThan {
+                                lt
+                            } else {
+                                LLVMBuildNot(ctx.builder, lt, c_str!(""))
+                            }
+                        }, 
+                        BinOpKind::GreaterThan | BinOpKind::LessThanOrEqual => {
+                            use super::cmp::GenCmp;
+                            let func = self.gen_cmp_fn(ty)?;
+                            let mut args = [ctx.get_value(left)?, ctx.get_value(right)?];
+                            let cmp = LLVMBuildCall(ctx.builder, func, args.as_mut_ptr(), args.len() as u32, c_str!(""));
+                            let gt = LLVMBuildICmp(ctx.builder, LLVMIntSGT, cmp,
+                                                   self.i32(0), c_str!(""));
+                            
+                            if op == BinOpKind::GreaterThan {
+                                gt
+                            } else {
+                                LLVMBuildNot(ctx.builder, gt, c_str!(""))
+                            }
+                        }, 
                         _ => unimplemented!(), // Vector comparison
```

Comment:
Remove comment and change to `unreachable!()` since all comparison cases are covered here.